core/src/main/java/io/questdb/cutlass/http/processors/PreferencesProcessor.java

```diff
@@ -0,0 +1,157 @@
+    private static final LocalValue<PreferencesProcessorState> LV = new LocalValue<>();
+    private static final Utf8String URL_PARAM_MODE = new Utf8String("mode");
+    private static final Utf8String URL_PARAM_VERSION = new Utf8String("version");
+    private final HttpFullFatServerConfiguration configuration;
+    private final PreferencesStore preferencesStore;
+    private final byte requiredAuthType;
+    private PreferencesStore.Mode mode;
+    private HttpConnectionContext transientContext;
+    private PreferencesProcessorState transientState;
+    private long version;
+
+    public PreferencesProcessor(CairoEngine engine, HttpFullFatServerConfiguration httpServerConfiguration) {
+        configuration = httpServerConfiguration;
+        preferencesStore = engine.getPreferencesStore();
+        requiredAuthType = configuration.getJsonQueryProcessorConfiguration().getRequiredAuthType();
+    }
+
+    @Override
+    public void close() {
+    }
+
+    @Override
+    public void failRequest(HttpConnectionContext context, HttpException e) throws PeerDisconnectedException, PeerIsSlowToReadException {
+        sendErr(e.getFlyweightMessage());
+    }
+
+    @Override
+    public byte getRequiredAuthType() {
+        return requiredAuthType;
+    }
+
+    @Override
+    public void onContent(long lo, long hi) throws PeerDisconnectedException, PeerIsSlowToReadException, ServerDisconnectException {
+        if (hi > lo) {
+            try {
+                transientState.sink.putNonAscii(lo, hi);
+            } catch (CairoException | CairoError e) {
+                sendErr(e.getFlyweightMessage());
+            }
+        }
+    }
+
+    @Override
+    public void onHeadersReady(HttpConnectionContext context) {
+        transientContext = context;
+        transientState = LV.get(context);
+        if (transientState == null) {
+            LOG.debug().$("new config state").$();
+            LV.set(context, transientState = new PreferencesProcessorState(configuration.getRecvBufferSize()));
+        }
+
+        mode = PreferencesStore.Mode.of(context.getRequestHeader().getUrlParam(URL_PARAM_MODE));
+
+        final Utf8Sequence versionStr = context.getRequestHeader().getUrlParam(URL_PARAM_VERSION);
+        try {
+            version = Numbers.parseLong(versionStr);
+        } catch (NumericException e) {
+            throw CairoException.nonCritical().put("Could not parse version, numeric value expected [version=").put(versionStr).put(']');
+        }
+    }
+
+    @Override
+    public void onRequestComplete(HttpConnectionContext context) throws PeerDisconnectedException, PeerIsSlowToReadException, ServerDisconnectException {
+        try {
+            context.getSecurityContext().authorizeSystemAdmin();
+
+            preferencesStore.save(transientState.sink, mode, version);
+            sendOk();
+        } catch (JsonException | CairoError e) {
+            LOG.error().$("error while saving config").$((Throwable) e).$();
```

Comment:
nit: terminology + formatting "could not" or "cannot" - for consistency

Reply:
addressed in https://github.com/questdb/questdb/pull/5621/commits/013292a54d52a81912bb2fc3205ce656f2d82f96